src/Tools/dotnet-monitor/CollectionRules/ExecuteActionValidateOptions.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using Microsoft.Extensions.Options;
+using System.IO;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules
+{
+    internal sealed class ExecuteActionValidateOptions : IValidateOptions<ExecuteOptions>
```

Comment:
This validator is not necessary as it is already covered by the Required attribute on the ExecuteOptions.Path property + the DataAnnotationValidatOptions implementation registered for these options.

Reply:
Also, the path existence should probably be checked in the IAction<T>.Execute method so that configuration can pass validation when its loaded.

Reply:
What do you mean by "so that configuration can pass validation when its loaded?" My understanding of what you want is just to perform a check upon entering `ExecuteAsync` (and then failing the action if there's nothing to execute for the path), but I just want to make sure I'm understanding what you're asking for.

Reply:
> What do you mean by "so that configuration can pass validation when its loaded?"

Just because the path doesn't exist when the options are loaded from configuration doesn't mean that the path won't exist when you go to execute the action. For example, if we are to perform token replacement for the options of each action before the action is executed, then you cannot verify that the path exists when you load the options (because replacing the tokens may make the path exist). Another example is what if the process of one Execute action writes out a file to disk in a known location and the subsequent Execute action has that file specified in its options? That file doesn't exist at options load time but only after executing the first action.

> My understanding of what you want is just to perform a check upon entering `ExecuteAsync` (and then failing the action if there's nothing to execute for the path), but I just want to make sure I'm understanding what you're asking for.

Yes, check that the path exits before calling `Process.Start` on it and throw a reasonable exception e.g. `FileNotFoundException`.